File: BPG-Bioland/ViewController.m
//
//  ViewController.m
//  BPG-Bioland
//
//  Created by starlueng on 2016/8/10.
//  Copyright © 2016年 starlueng. All rights reserved.
//

#import "ViewController.h"
#import "BPGProtocol.h"
@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    [[BPGProtocol shareInstance]babyDelegate];
    
    NSArray *titles = @[@"开机",@"关机",@"上传一次数据",@"上传所有数据",@"开始测量",@"停止测量",@"设备时间改变",@"脉冲设置"];
    for (NSInteger i =0; i<8; i++) {
        UIButton *button = [UIButton buttonWithType:UIButtonTypeSystem];
        [button setTitle:titles[i] forState:UIControlStateNormal];
        [button addTarget:self action:@selector(click:) forControlEvents:UIControlEventTouchUpInside];
        button.frame = CGRectMake(100 *(i%3), 100*(i/3)+100, 100, 100);
        button.tag = 100+i;
        [self.view addSubview:button];
    }
  
    // Do any additional setup after loading the view, typically from a nib.
}
- (void) click:(UIButton *)sender{
    switch (sender.tag) {
        case 100:
        {
            [[BPGProtocol shareInstance]turnOnModeWith:^(BPGDataModel *model) {
                NSLog(@"%@",model);
            } OrFail:^(NSError *error) {
                NSLog(@"%@",error);
            }];
            [[BPGProtocol shareInstance]realTestMode:^(BPGDataModel *model) {
                NSLog(@"%@",model);
            } OrFail:^(NSError *error) {
                NSLog(@"%@",error);
            }];

        }
            break;
        case 101:
        {
            [[BPGProtocol shareInstance]turnOffModeWith:^(BPGDataModel *model) {
                NSLog(@"%@",model);
            } OrFail:^(NSError *error) {
                NSLog(@"%@",error);
            }];
        }
            break;
        case 102:
        {
            [[BPGProtocol shareInstance]getLastDataWith:^(BPGDataModel *model) {
                
                NSLog(@"%@",model);
            } OrFail:^(NSError *error) {
                
                NSLog(@"%@",error);
            }];
        }
            break;
        case 103:
        {
            [[BPGProtocol shareInstance]getAllTestDataWith:^(BPGDataModel *model) {
                NSLog(@"%@",model);
            } OrFail:^(NSError *error) {
                NSLog(@"%@",error);
            }];
        }
            break;
        case 104:
        {
            [[BPGProtocol shareInstance]realTestMode:^(BPGDataModel *model) {
                NSLog(@"%@",model);
            } OrFail:^(NSError *error) {
                NSLog(@"%@",error);
            }];
        }
            break;
        case 105:
        {
            [[BPGProtocol shareInstance]pauseMode:^(BPGDataModel *model) {
                NSLog(@"%@",model);
            } OrFail:^(NSError *error) {
                NSLog(@"%@",error);
            }];
        }
            break;
        case 106:
        {
            
            [[BPGProtocol shareInstance]setTimeWithTimeData:nil result:^(BPGDataModel *model) {
                
                NSLog(@"%@",model);
            } OrFail:^(NSError *error) {
                  NSLog(@"%@",error);
            }];
                    }
            break;
        case 107:
        {
            [[BPGProtocol shareInstance]isSendDeployWith:^(BPGDataModel *model) {
                NSLog(@"%@",model);
            } OrFail:^(NSError *error) {
                NSLog(@"%@",error);
            }];
        }
            break;
        default:
            break;
    }
    
   
    

}
- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: BPGBiolands/ViewController.swift
//
//  ViewController.swift
//  BPGBiolands
//
//  Created by starlueng on 2016/8/12.
//  Copyright © 2016年 starlueng. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
    
    var  BPGPro :BPGProtocol = BPGProtocol.init()
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        BPGPro.BLEDelegate()
        let titles = ["开机","关机","开测","暂停"]
        
        for i in 0 ..< 4 {
            let button = UIButton(type:.system)
            button.frame = CGRect(x: i / 2 * 100 ,y:100 + i % 2 * 100 ,width: 100 ,height :100)
            button.setTitle(titles[i], for: .normal)
            button.addTarget(self, action: #selector(ViewController.actions(button:)), for: .touchUpInside)
            button.tag = 10 + i
            self.view.addSubview(button)
        }
        // Do any additional setup after loading the view, typically from a nib.
    }
   @objc func actions(button:UIButton) {
    
        switch button.tag {
        case 10:
            BPGPro.turnOnModeWith(success: { (model :BPGModel?) in
                
                print("\(model)")
                }, fail: { (error :NSError?) in
                    print("\(error)")
            })
                
        case 11:
            BPGPro.turnOffModeWith(success: { (model :BPGModel?) in
                
                print("\(model)")
                }, fail: { (error :NSError?) in
                    print("\(error)")
            })
        case 12:
            BPGPro.realTestMode(success: { (model :BPGModel?) in
                
                print("\(model)")
                }, fail: { (error :NSError?) in
                    print("\(error)")
            })
        case 13:
            BPGPro.pauseMode(success: { (model :BPGModel?) in
                
                print("\(model)")
                }, fail: { (error :NSError?) in
                    print("\(error)")
            })
        default:
            break
        }
    }
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

